packages/next-auth/src/lib/actions.ts

```diff
@@ -73,8 +71,11 @@ export async function signIn(
 
   for (const c of res?.cookies ?? []) cookies().set(c.name, c.value, c.options)
 
-  if (shouldRedirect) return redirect(res.redirect!)
-  return res.redirect as any
+  const responseUrl =
+    res instanceof Response ? res.headers.get("Location") : res.redirect
+
+  if (shouldRedirect) return redirect(responseUrl!)
```

Comment:
Agree but what should happen when `undefined`?

Reply:
I mean, what should happen here? What should dev using next-auth expect when this is `undefined`.

To be clear. It always is possible in the future for this to be `undefined`. But as far as my understanding of the code with this fix is not possible.

Reply:
So we landed here because an error is thrown somewhere in `AuthInternal` - in this case its the `sendVerificationToken`. I think it'll be logical to redirect to the `signin` page. It's the same behavior of the signin action in core when missing the provider:
https://github.com/nextauthjs/next-auth/blob/770112c7a12c30f9a294bd8e0bbaef5274c4ad4f/packages/core/src/lib/actions/signin/index.ts#L18

I think we can do `return redirect(url.toString())`